subworkflows/local/utils_nfcore_seqinspector_pipeline/main.nf

```diff
@@ -84,12 +84,20 @@ workflow PIPELINE_INITIALISATION {
         .fromSamplesheet("input") // Validates samplesheet against $projectDir/assets/schema_input.json. Path to validation schema is defined by $projectDir/nextflow_schema.json
         .map {
             meta, fastq_1, fastq_2 ->
-                def tags = meta.tags ? meta.tags.tokenize(",") : []
+                def tags = meta.tags ? meta.tags.tokenize(":") : []
                 def updated_meta = meta + [ id:meta.sample, tags:tags ]
                 if (!fastq_2) {
-                    return [ updated_meta.id, updated_meta + [ single_end:true ], [ fastq_1 ] ]
+                    return [
+                        updated_meta.id + fastq_1.toString().replaceAll('/', '_'),
```

Comment:
What is the rationale to attach the file path(s) to the `meta.id` from the sample sheet?

Reply:
This is related to the question I had about making the ids unique. If a sample is run e.g. on two different lanes, two rows will have the same id. This is my implementation of @kedhammar 's second alternative (see above: https://github.com/nf-core/seqinspector/pull/20#issuecomment-2312738755)

Reply:
@MatthiasZepper we are kind of heading to a direction where the fastq path is the only unique property of each row, so I thought it made sense to make it the dataframe index, if you will.

Reply:
Ah, ok, that is indeed a valid concern. But can't we just enumerate the lines of the sample sheet or something? `fastq_1.toString().replaceAll('/', '_')` feels quite massive...

Reply:
Maybe not deterministic...because the order of the channel may change with a real `.fromSamplesheet()` factory?!? 

```
    ch_fruit = Channel.from("apple","mango", "banana", "banana","peach")
    ch_fruit_unique = ch_fruit.toList().flatMap { it.withIndex().collect{ entry, idx -> "${entry}_${idx}" }}
    ch_fruit_unique.view()
```
which prints
```
apple_0
mango_1
banana_2
banana_3
peach_4
```

Reply:
I would prefer a simple number suffix being added to the `meta.sample` to form the `meta.id` :

```

Channel.fromSamplesheet("input")
    .toList()
    .flatMap{it.withIndex().collect{ entry, idx -> entry + "${idx+1}" }}
    .map { meta, fastq_1, fastq_2, idx -> 
        def tags = meta.tags ? meta.tags.tokenize(":") : []
        def updated_meta = meta + [ id: "${meta.sample}_${idx}", tags:tags, single: fastq_2[0] ? true : false ] 
        return [updated_meta.id, updated_meta, fastq_1 , fastq_2]
    }.view()
```

Reply:
I've asked for guidance on the nf-core slack (https://nfcore.slack.com/archives/C043FMKUNLB/p1725872244960039), from what I understand `id` can be useful to track down samples when debugging. So I'd follow James' suggestion and just use filenames instead of full paths.